src/clj/rems/workflow/dynamic.clj

```diff
@@ -75,6 +76,17 @@
          :event/type (s/eq :application.event/comment-requested)
          :application/commenters [s/Str]
          :application/comment s/Str))
+(s/defschema CreatedEvent
+  (assoc EventBase
+         :event/type (s/eq :application.event/created)
+         :application/resources [{:catalogue-item/id s/Int
+                                  :resource/ext-id s/Str}]
+         :application/licenses [{:license/id s/Int}]
+         :form/id s/Int
+         :workflow/id s/Int
+         ;; TODO: separate workflow specific data to a new event or make this an open schema?
```

Comment:
Why are this values at the top level? Could it be just `:workflow s/Any`  or alternatively something more specific under the `:workflow`?

Reply:
With [namespaced keys](https://github.com/CSCfi/rems/blob/master/docs/architecture/003-namespaced-keywords.md) they don't need be nested, but each workflow can add its own workflow-specific stuff (in this case only `:workflow.dynamic/handlers`) using its own namespace. The `:workflow.dynamic/handlers` key should probably be removed from this event schema and replaced with `s/Keyword s/Any`.

Reply:
Or actually, removing `:workflow.dynamic/handlers` would prevent using coercion when deserializing the events. Making the key optional seems better.